### PR TITLE
feat: DTOS-3638 Parquet file functionality Added

### PR DESCRIPTION
**Pull request by sudipnhs**
## Type of changes

- [ ] Refactoring (non-breaking change)
- [ ] New feature (non-breaking change which adds functionality)
- [x] Breaking change (fix or feature that would change existing functionality)
- [ ] Bug fix (non-breaking change which fixes an issue)

## Checklist

- [ ] I am familiar with the [contributing guidelines](../docs/CONTRIBUTING.md)
- [x] I have followed the code style of the project
- [x] I have added tests to cover my changes
- [ ] I have updated the documentation accordingly
- [ ] This PR is a result of pair or mob programming

---

## Sensitive Information Declaration

To ensure the utmost confidentiality and protect your and others privacy, we kindly ask you to NOT including [PII (Personal Identifiable Information) / PID (Personal Identifiable Data)](https://digital.nhs.uk/data-and-information/keeping-data-safe-and-benefitting-the-public) or any other sensitive data in this PR (Pull Request) and the codebase changes. We will remove any PR that do contain any sensitive information. We really appreciate your cooperation in this matter.

- [x] I confirm that neither PII/PID nor sensitive data are included in this PR and the codebase changes.

### CODE REVIEW

File: tests/ScreeningValidationServiceTests/LookupValidation/LookupValidationTests.cs [hunk 1]
```diff
@@ -310,7 +310,7 @@ public async Task Run_invalidParticipant_ValidateBsoCodeRuleFails(string ReasonF
         _requestBody.NewParticipant.PrimaryCareProvider = primaryCareProvider;
         _requestBody.NewParticipant.Postcode = postcode;
         _requestBody.NewParticipant.ReasonForRemoval = ReasonForRemoval;
-        _requestBody.NewParticipant.RecordType = "New";
```
Comment (andrewmorris43) [inline tests/ScreeningValidationServiceTests/LookupValidation/LookupValidationTests.cs:313]: Can this be Actions.New instead?

File: tests/ScreeningValidationServiceTests/StaticValidation/StaticValidationTests.cs
Comment (andrewmorris43) [inline tests/ScreeningValidationServiceTests/StaticValidation/StaticValidationTests.cs:1]: I know all these recordTypes were already hardcoded, but could we change them to use the Actions constant instead?

File: tests/ScreeningValidationServiceTests/LookupValidation/LookupValidationTests.cs (hunk 1 above)
Comment (sudipnhs) [reply tests/ScreeningValidationServiceTests/LookupValidation/LookupValidationTests.cs:313]: I have changed _requestBody.NewParticipant.RecordType = "New" ---to-- _requestBody.NewParticipant.RecordType = "ADD" in order to match new action name in parquet file.
Yes, you are right, Actions.New can be done instead but for this work we will be creating a new tickets for refactoring the validation code in the entire project.
So for now, we will keep it like this as it's not part of this parquet functionality ticket

File: tests/ScreeningValidationServiceTests/StaticValidation/StaticValidationTests.cs
Comment (sudipnhs) [reply tests/ScreeningValidationServiceTests/StaticValidation/StaticValidationTests.cs:1]: I agree Andrew. As i said, we will have a separate ticket for this. This ticket is going on for very long time.
If we try to fix all issue in one ticket then we cannot close the ticket.

Comment (sudipnhs) [reply tests/ScreeningValidationServiceTests/StaticValidation/StaticValidationTests.cs:1]: Yes, I agree Andrew. As i said, we will have a separate ticket for this. This ticket is going on for very long time.
If we try to fix all issue in one ticket then we cannot close the ticket.

File: application/CohortManager/src/Functions/Shared/Model/Model.csproj [hunk 2]
```diff
@@ -7,7 +7,7 @@
   </PropertyGroup>
 
   <ItemGroup>
-    <PackageReference Include="CsvHelper" Version="32.0.3" />
+    <PackageReference Include="ParquetSharp" Version="17.0.0-beta1" />
```
[the remark this answers is not in the document]
Comment (sudipnhs) [reply application/CohortManager/src/Functions/Shared/Model/Model.csproj:10]: yes , it's sufficient and working fine. We have tested it locally.

**Review by andrewmorris43 — APPROVED** [review]
I'm happy that, looks good to me

**Review by MWClayson-NHS — APPROVED** [review]

**Pull request merged**